eq_translations/survey_schema.py

```diff
@@ -1,245 +1,219 @@
-
-    @property
-    def no_context_pointers(self):
-        return (
-            self.get_core_pointers()
-            + self.get_title_pointers()
-            + self.get_message_pointers()
-            + self.get_list_pointers()
-            + self.no_context_placeholder_pointers
-        )
+        with open(schema_path, "w", encoding="utf8") as schema_file:  # pragma: no cover
+            return json.dump(self.schema, schema_file, indent=4)  # pragma: no cover
 
     @property
-    def context_pointers(self):
-        return self.get_answer_pointers() + self.context_placeholder_pointers
-
-    def get_core_pointers(self):
+    def translatable_strings(self):
         """
-        Iterate the schema and return the pointers found
-        :return:
+        :return: A list of pointer self._get_pointer_dict
         """
         pointers = []
-        for key in SurveySchema.no_context_keys:
-            key_pointers = find_pointers_to(self.schema, key)
-            pointers.extend(key_pointers)
-        return pointers
 
-    def get_placeholder_pointers(self):
-        """
-        Placeholder pointers may have context or may not
-        :return:
-        """
-        found_pointers = find_pointers_to(self.schema, "text")
-        context_placeholder_pointers = []
-        no_context_placeholder_pointers = []
+        for key in self.keys_to_translate:
+            with_context = key in self.keys_with_context
 
-        for pointer in found_pointers:
-            if "/answers/" in pointer:
-                context_placeholder_pointers.append(pointer)
-            else:
-                no_context_placeholder_pointers.append(pointer)
+            for pointer in find_pointers_to(self.schema, key):
+                # The 'list' property is used by both 'contents' and 'when' blocks.
+                if "/when/" in pointer:
+                    continue
 
-        return context_placeholder_pointers, no_context_placeholder_pointers
+                schema_element = resolve_pointer(self.schema, pointer)
+                if isinstance(schema_element, dict):
+                    pointers_found = self._get_pointers_from_dict_element(
+                        pointer, schema_element, with_context
+                    )
+                    pointers.extend(pointers_found)
 
-    def get_message_pointers(self):
-        """
-        Message pointers need to be iterated as a dict and each key added individually
-        :return:
-        """
-        pointers = []
+                elif isinstance(schema_element, list):
+                    pointers_found = self._get_pointers_from_list_element(
+                        pointer, schema_element, with_context
+                    )
+                    pointers.extend(pointers_found)
+
+                else:
+                    pointer_dict = self._get_pointer_dict(
+                        pointer, schema_element, with_context
+                    )
+                    pointers.append(pointer_dict)
```

Comment:
This whole pattern of nested appending to dicts can be avoided by using yield

Reply:
As discussed, this now uses yield and have also removed the repeated called to the property when using `len()` to get total messages.